src/Tribe/Live_Content.php

```diff
@@ -0,0 +1,422 @@
+		$arguments = $this->get_arguments();
+		$argument  = Arr::get( $arguments, $index, $default );
+
+		/**
+		 * Applies a filter to a specific shortcode argument, catch all for all shortcodes..
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed  $argument The argument.
+		 * @param array  $index    Which index we indent to fetch from the arguments.
+		 * @param array  $default  Default value if it doesnt exist.
+		 * @param static $instance Which instance of shortcode we are dealing with.
+		 */
+		$argument = apply_filters( 'tribe_ext_shortcode_argument', $argument, $index, $default, $this );
+
+		$registration_slug = $this->get_registration_slug();
+
+		/**
+		 * Applies a filter to a specific shortcode argument, to a particular registration slug.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed  $argument The argument value.
+		 * @param array  $index    Which index we indent to fetch from the arguments.
+		 * @param array  $default  Default value if it doesnt exist.
+		 * @param static $instance Which instance of shortcode we are dealing with.
+		 */
+		$argument = apply_filters( "tribe_ext_shortcode_{$registration_slug}_argument", $argument, $index, $default, $this );
+
+		return $argument;
+	}
+
+	/**
+	 * Returns a shortcode default arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_default_arguments() {
+		/**
+		 * Applies a filter to instance default arguments.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $default_arguments Current set of default arguments.
+		 * @param static $instance          Which instance of shortcode we are dealing with.
+		 */
+		$default_arguments = apply_filters( 'tribe_ext_shortcode_default_arguments', $this->default_arguments, $this );
+
+		$registration_slug = $this->get_registration_slug();
+
+		/**
+		 * Applies a filter to instance default arguments based on the registration slug of the shortcode.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $default_arguments Current set of default arguments.
+		 * @param static $instance          Which instance of shortcode we are dealing with.
+		 */
+		$default_arguments = apply_filters( "tribe_ext_shortcode_{$registration_slug}_default_arguments", $this->default_arguments, $this );
+
+		return $default_arguments;
+	}
+
+	/**
+	 * Validation of Null or Truthy values for Shortcode Attributes.
+	 *
+	 * @since TBD
+	 *
+	 * @param mixed $value Which value will be validated.
+	 *
+	 * @return bool|null   Allows Both Null and truthy values.
+	 */
+	public static function validate_null_or_truthy( $value = null ) {
+		if ( null === $value || 'null' === $value ) {
+			return null;
+		}
+
+		return tribe_is_truthy( $value );
+	}
+
+	/**
+	 * Check if Post ID is set or set the current post id
+	 *
+	 * @since TBD
+	 *
+	 * @param $id
+	 *
+	 * @return false|int
+	 */
+	public static function validate_event( $id ) {
+		return ( null === $id || 'null' === $id ) ? get_the_ID() : $id;
+	}
```

Comment:
Consider using `tribe_post_exists()`

If null, get current post. If not null, check if post exists - if it doesn't (bad ID passed), have the shortcode return an empty string.